Jenkinsfile

```diff
@@ -0,0 +1,267 @@
+#!groovy
+
+def is_master_build(String job_name) {
+  // Master builds start with the platform, branch builds start with
+  // 'Branch-' or 'PR-'
+  if (job_name.tokenize('-')[0] in ['Branch', 'PR']) {
+    return false
+  } else {
+    return true
+  }
+}
+
+def get_platform(String job_name) {
+  // Get name of the platform e.g. 'Scientific-Linux-7'
+  def idxi
+  if (is_master_build(job_name)) {
+    idxi = 0
+  } else {
+    idxi = 1
+  }
+  return job_name.tokenize('-')[idxi..-2].join('-')
+}
+
+def get_matlab_version(String job_name) {
+  return job_name[-5..-1]
+}
+
+def get_agent(String job_name) {
+  if (get_platform(job_name) == 'Scientific-Linux-7') {
+    withCredentials([string(credentialsId: 'sl7_agent', variable: 'agent')]) {
+      return "${agent}"
+    }
+  } else if (get_platform(job_name) == 'Windows-10') {
+    withCredentials([string(credentialsId: 'win10_agent', variable: 'agent')]) {
+      return "${agent}"
+    }
+  } else {
+    return ''
+  }
+}
+
+def get_build_info(String repo, String branch, String match_context) {
+  def job_name
+  def build_num
+  def script_cmd = "python get_build_info.py ${repo} ${branch}"
+  if (match_context) {
+    script_cmd += " --match-context ${match_context}"
+  }
+  if (isUnix()) {
+    build_info = sh(script: "module load conda/3 && ${script_cmd}", returnStdout: true)
+  } else {
+    build_info = bat(script: script_cmd, returnStdout: true)
+  }
+  println build_info
+  // Index from the end to ignore any previous output
+  job_name = build_info.tokenize(' |\n')[-2].trim()
+  build_num = build_info.tokenize(' |\n')[-1].trim()
+  return [job_name, build_num]
+}
+
+properties([
+  parameters([
+    string(
+      defaultValue: get_matlab_version(env.JOB_BASE_NAME),
+      description: 'The version of Matlab to use e.g. 2019b.',
+      name: 'MATLAB_VERSION',
+      trim: true
+    ),
```

Comment:
We've recently started declaring our parameter defaults differently, due to some issues when changing them. Have a look at this issue https://github.com/pace-neutrons/horace/issues/574. The issue boils down to the defaults only being updated _after_ the pipeline starts.

These are the PRs for Herbert and Horace: https://github.com/pace-neutrons/Herbert/pull/307 https://github.com/pace-neutrons/Horace/pull/587 resolving said issue.